controller-server/src/main/java/com/yahoo/vespa/hosted/controller/LockedApplication.java

```diff
@@ -126,42 +128,29 @@ public LockedApplication withOutstandingChange(boolean outstandingChange) {
                                                      deploymentJobs(), deploying(), outstandingChange), lock);
     }
 
-    private Version determineTriggerVersion(DeploymentJobs.JobType jobType, Controller controller) {
-        Optional<Zone> zone = jobType.zone(controller.system());
-        if ( ! zone.isPresent()) // a sloppy test TODO: Fix
-            return controller.systemVersion();
-        return currentDeployVersion(controller, zone.get());
+    private Version deployVersionFor(DeploymentJobs.JobType jobType, Controller controller) {
+        return jobType == JobType.component
+               ? controller.systemVersion()
+               : deployVersionIn(jobType.zone(controller.system()).get(), controller);
```

Comment:
I believe these conditions are the same? The only JobType we ever trigger, which does not have a zone, is the component job, unless we are doing something very wrong.